File: src/main/java/com/back/nbe9112team06/global/response/ApiResponse.java
package com.back.nbe9112team06.global.response;

import com.fasterxml.jackson.annotation.JsonIgnore;

public record ApiResponse<T>(
        String resultCode,
        String msg,
        T data
) {
    public ApiResponse(String msg, String resultCode){
        this(resultCode, msg,null);
    }

    @JsonIgnore
    public int getStatusCode() {
        return Integer.parseInt(resultCode.split("-")[0]);
    }
}

File: src/main/kotlin/com/back/nbe9112team06/global/response/ApiResponse.kt
package com.back.nbe9112team06.global.response

import com.fasterxml.jackson.annotation.JsonIgnore

data class ApiResponse<T>(
    val resultCode: String,
    val msg: String,
    val data: T? = null
) {
    constructor(msg: String, resultCode: String) : this(resultCode, msg, null)

    //TODO substringBefore로 바꾸긴했지만 추후에 다시 확인 원래는 split으로 나누고 있었다.
    @get:JsonIgnore
    val statusCode: Int
        get() = resultCode
            .substringBefore("-", resultCode)
            .toInt()
}